### ISSUE TITLE
Type short name not recognized in an update operation.

### ISSUE THREAD

**Issue by jason-fox** [Fiware/context.Orion-LD#507]
I create three Data Entities:

```console
curl -L -X POST 'http://localhost:1026/ngsi-ld/v1/entityOperations/create' \
-H 'Content-Type: application/json' \
-H 'Link: <http://context-provider:3000/data-models/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-H 'Accept: application/ld+json' \
-H 'Cookie: _csrf=MAPTGFPcoPnewsGCWklHi4Mq' \
--data-raw '[
    {
      "id": "urn:ngsi-ld:TemperatureSensor:002",
      "type": "TemperatureSensor",
      "category": {
            "type": "Property",
            "value": "sensor"
      },
      "temperature": {
            "type": "Property",
            "value": 20,
            "unitCode": "CEL"
      }
    },
    {
      "id": "urn:ngsi-ld:TemperatureSensor:003",
      "type": "TemperatureSensor",
      "category": {
            "type": "Property",
            "value": "sensor"
      },
      "temperature": {
            "type": "Property",
            "value": 2,
            "unitCode": "CEL"
      }
    },
     {
      "id": "urn:ngsi-ld:TemperatureSensor:004",
      "type": "TemperatureSensor",
      "category": {
            "type": "Property",
            "value": "sensor"
      },
      "temperature": {
            "type": "Property",
            "value": 100,
            "unitCode": "CEL"
      }
    }
]'
```

I then update two of them using an `update` operation as shown:

```console
curl -L -X POST 'http://localhost:1026/ngsi-ld/v1/entityOperations/update?options=replace' \
-H 'Content-Type: application/json' \
-H 'Link: <http://context-provider:3000/data-models/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
-H 'Cookie: _csrf=MAPTGFPcoPnewsGCWklHi4Mq' \
--data-raw '[
  {
    "id": "urn:ngsi-ld:TemperatureSensor:003",
    "type": "https://uri.fiware.org/ns/data-models#TemperatureSensor",
    "category": {
      "type": "Property",
      "value": [
        "actuator",
        "sensor"
      ]
    }
  },
  {
    "id": "urn:ngsi-ld:TemperatureSensor:004",
    "type": "TemperatureSensor",
    "temperature": {
      "type": "Property",
      "value": [
        "actuator",
        "sensor"
      ]
    }
  }
]'
```

Both `TemperatureSensor:003` and `TemperatureSensor:004` exist so the expected result is **200 Success**, but the actual response is:  **207 Multi Status**

```jsonld
{
    "@context": "http://context-provider:3000/data-models/ngsi-context.jsonld",
    "success": [
        "urn:ngsi-ld:TemperatureSensor:003"
    ],
    "errors": [
        {
            "entityId": "urn:ngsi-ld:TemperatureSensor:004",
            "error": {
                "type": "https://uri.etsi.org/ngsi-ld/errors/BadRequestData",
                "title": "non-matching entity type",
                "detail": "TemperatureSensor",
                "status": 400
            }
        }
    ]
}
```

```
"git_hash": "db891a72a531ef4cc037caed5b04f8fe71da6814",
"compile_time": "Thu Jul 30 13:13:23 UTC 2020"
```

**Comment by jason-fox** [comment]
The `@context` used in the example can be found [here](https://raw.githubusercontent.com/FIWARE/tutorials.NGSI-LD/master/app/public/data-models/ngsi-context.jsonld).

**Comment by jason-fox** [comment]
The same operation as an `upsert` works as expected - **204 No Content**

**POST** `http://localhost:1026/ngsi-ld/v1/entityOperations/upsert?options=replace`

Although no body is found in the response - see https://github.com/FIWARE/context.Orion-LD/issues/502

**Comment by kzangeli** [comment]
Fixed by PR #513

**Comment by jason-fox** [comment]
Verified: eddd37530fb7cb748ad582b5c20ea4327accd558
